onnxruntime/core/mlas/lib/erf.cpp

```diff
@@ -0,0 +1,273 @@
+        r_small = MlasAndNotFloat32x4(split_mask, r_small);
+
+        MLAS_FLOAT32X4 r_big = MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P0);
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P1));
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P2));
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P3));
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P4));
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P5));
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, MlasBroadcastFloat32x4(MlasErfConstants.ErfBIG_P6_Minus_One));
+        r_big = MlasMultiplyAddFloat32x4(r_big, AbsValue, AbsValue);
+
+        // 1.0 - exp(-r_big), no need to do min()
+        r_big = MlasXorFloat32x4(r_big, MlasBroadcastFloat32x4(MlasErfConstants.ErfNegZero)); // -r_big
+        r_big = MlasMaximumFloat32x4(MlasBroadcastFloat32x4(MlasErfConstants.Exp_LowerRange), r_big);
+        MLAS_FLOAT32X4 exp_c = MlasBroadcastFloat32x4(MlasErfConstants.Exp_C);
+        MLAS_FLOAT32X4 r = MlasMultiplyAddFloat32x4(MlasBroadcastFloat32x4(MlasErfConstants.Exp_Log2Reciprocal), r_big, exp_c);
+        r = MlasSubtractFloat32x4(r, exp_c);
+
+        MLAS_FLOAT32X4 fx = MlasMultiplyAddFloat32x4(r, MlasBroadcastFloat32x4(MlasErfConstants.Exp_log2_hi), r_big);
+        fx = MlasMultiplyAddFloat32x4(r, MlasBroadcastFloat32x4(MlasErfConstants.Exp_log2_lo), fx);
+        // y = exp(fx)
+        MLAS_FLOAT32X4 y = MlasBroadcastFloat32x4(MlasErfConstants.Exp_P0);
+        y = MlasMultiplyAddFloat32x4(y, fx, MlasBroadcastFloat32x4(MlasErfConstants.Exp_P1));
+        y = MlasMultiplyAddFloat32x4(y, fx, MlasBroadcastFloat32x4(MlasErfConstants.Exp_P2));
+        y = MlasMultiplyAddFloat32x4(y, fx, MlasBroadcastFloat32x4(MlasErfConstants.Exp_P3));
+        y = MlasMultiplyAddFloat32x4(y, fx, MlasBroadcastFloat32x4(MlasErfConstants.Exp_P4));
+        y = MlasMultiplyAddFloat32x4(y, fx, MlasBroadcastFloat32x4(MlasErfConstants.Exp_P5));
+        y = MlasMultiplyAddFloat32x4(y, fx, MlasBroadcastFloat32x4(MlasErfConstants.Exp_P6));
+        // 1.0 - exp(fx) * 2^INT(r)
+        y = MlasMultiplyFloat32x4(y, MlasPowerOf2Float32x4(r));
+        y = MlasSubtractFloat32x4(MlasBroadcastFloat32x4(MlasErfConstants.ErfOne), y);
+
+        // merge two splits results
+        y = MlasOrFloat32x4(r_small, y);
+        y = MlasOrFloat32x4(y, SignMask);
+
+        MlasStoreFloat32x4(Output, y);
+
+        Input += 4;
+        Output += 4;
+        N -= 4;
+    }
+
+    while (N > 0) {
+        float Value = *Input++;
+        float AbsValue = fabsf(Value);
+
+        float r;
+        if (AbsValue > MlasErfConstants.ErfSplitBoundary) {
+            AbsValue = (std::min)(MlasErfConstants.ErfUpperAbsRange, AbsValue);
```

Comment:
How much of a deviation do you see from results produced the vectorized way vs the scalar path?

Reply:
Just used small set of value to test, result is exactly same.
I feel the code logic is also exactly same.

Reply:
Yeah, you are right. famf cause slightly different value than those in vectorized code. Change them all to * and +.
Also fixed errors in logic on different branch with more test data and onnx big test data tested manually.